### PR TITLE
Fix directConnect

### PR DESCRIPTION
**Pull request by KazuCocoa**
## Proposed changes

Fixes https://github.com/webdriverio/webdriverio/issues/7427

The behavior was changed in https://github.com/webdriverio/webdriverio/pull/5853. This PR tries to bring the feature back to be expected. cc @jlipps 

After some testing, it needed to update `client.options` rather than modify `params` to apply hostnames etc correctly.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

Let me try to run tests on my local... then will tweak documentation.

### CODE REVIEW

File: packages/wdio-types/src/Capabilities.ts [hunk 1]
```diff
@@ -367,6 +367,13 @@ export interface AppiumCapabilities {
     eventTimings?: boolean;
     enablePerformanceLogging?: boolean;
     printPageSourceOnFindFailure?: boolean;
+
+    // Users as directConnect feature by the server
+    // https://appiumpro.com/editions/86-connecting-directly-to-appium-hosts-in-distributed-environments
+    directConnectProtocol?: string;
+    directConnectHost?: string;
+    directConnectPort?: number;
+    directConnectPath?: string;
```
Comment (christian-bromann) [inline packages/wdio-types/src/Capabilities.ts:376]: Can we also add them to `AppiumW3CCapabilities`?

Comment (christian-bromann) [reply packages/wdio-types/src/Capabilities.ts:376]: In setup direct connect we should then do:

```js
const directConnectProtocol = params.capabilities.directConnectProtocol || params.capabilities['appium:directConnectProtocol']
```

File: packages/wdio-types/src/Options.ts [hunk 2]
```diff
@@ -183,10 +183,7 @@ export interface WebDriver extends Connection {
     /**
      * Appium direct connect options (see: https://appiumpro.com/editions/86-connecting-directly-to-appium-hosts-in-distributed-environments)
      */
-    directConnectProtocol?: string
-    directConnectHost?: string
-    directConnectPort?: number
-    directConnectPath?: string
+    enableDirectConnect?: boolean;
```
Comment (christian-bromann) [inline packages/wdio-types/src/Options.ts:186]: Can we document this option [here](https://github.com/webdriverio/webdriverio/blob/main/website/docs/Options.md#webdriver-options)?

**Review by christian-bromann — COMMENTED** [review]
Thanks for fixing this 🙏 

Some suggestions, otherwise LGTM

**Comment by KazuCocoa** [comment]
@christian-bromann Could you review again when you have time?
I've tested that current implementation worked expectedly as https://appiumpro.com/editions/86-connecting-directly-to-appium-hosts-in-distributed-environments

Btw, does the e2e-mac work...?
The scenario worked when I modified the config and expected value as localhost or a host, but it is failing.
(As of current implementation, this directConnect should not affect the test case though..)

**Review by christian-bromann — APPROVED** [review]
Awesome 👍  LGTM

The e2e tests are very flaky atm, I haven't had the chance to look into them. We can just move forward with this.

**Pull request merged**